jacsstorage-services/src/main/java/org/janelia/jacsstorage/service/s3/S3Adapter.java

```diff
@@ -0,0 +1,147 @@
+        }
+        asyncS3ClientBuilder.credentialsProvider(credentialsProvider);
+        return asyncS3ClientBuilder
+                .crossRegionAccessEnabled(true)
+                .forcePathStyle(s3Options.getPathStyleBucket())
+                .multipartEnabled(true)
+                .build();
+    }
+
+    private S3Client createSyncClient(String endpoint, AwsCredentialsProvider credentialsProvider, JADEOptions s3Options) {
+        S3ClientBuilder s3ClientBuilder = S3Client.builder();
+        if (StringUtils.isNotBlank(s3Options.getAWSRegion())) {
+            Region s3Region = Region.of(s3Options.getAWSRegion());
+            s3ClientBuilder.region(s3Region);
+        }
+        if (StringUtils.isNotBlank(endpoint)) {
+            URI endpointURI = URI.create(endpoint);
+            s3ClientBuilder.endpointOverride(endpointURI);
+        }
+        s3ClientBuilder.credentialsProvider(credentialsProvider);
+
+        S3Configuration s3Configuration = S3Configuration.builder()
+                .multiRegionEnabled(true)
+                .pathStyleAccessEnabled(s3Options.getPathStyleBucket())
+                .build();
+
+        return s3ClientBuilder
+                .serviceConfiguration(s3Configuration)
+                .build();
+    }
+
+    private AwsCredentialsProvider createCredentialsProvider(JADEOptions s3Options) {
+        AwsCredentialsProvider credentialsProvider;
+        // when credentials (AWS accessKey and secretKey) are provided we only use the static credentials provider
+        // otherwise we chain multiple providers
+        if (StringUtils.isNotBlank(s3Options.getAccessKey()) && StringUtils.isNotBlank(s3Options.getSecretKey())) {
+            credentialsProvider = StaticCredentialsProvider.create(AwsBasicCredentials.create(s3Options.getAccessKey(), s3Options.getSecretKey()));
+        } else {
+            AwsCredentialsProviderChain.Builder credentialsProviderBuilder = AwsCredentialsProviderChain.builder();
+            boolean tryAnonymousAccessFirst = isTryAnonymousFirst(s3Options);
+            if (tryAnonymousAccessFirst) {
+                credentialsProviderBuilder.addCredentialsProvider(AnonymousCredentialsProvider.create());
+            }
+            credentialsProviderBuilder
+                    .addCredentialsProvider(ProfileCredentialsProvider.create())
+                    .addCredentialsProvider(SystemPropertyCredentialsProvider.create())
+                    .addCredentialsProvider(EnvironmentVariableCredentialsProvider.create())
+                    .addCredentialsProvider(InstanceProfileCredentialsProvider.create());
+            if (!tryAnonymousAccessFirst) {
+                credentialsProviderBuilder.addCredentialsProvider(AnonymousCredentialsProvider.create());
+            }
+            credentialsProvider = credentialsProviderBuilder.build();
+        }
+        return credentialsProvider;
+    }
+
+    private boolean isTryAnonymousFirst(JADEOptions s3Options) {
```

Comment:
it would be nice to explain options like this. What does it mean to try anonymous first? Why would you want to set this? It's not documented here or in JADEOptions.

Reply:
This is something that should be removed in the end and leave the anonymous credentials always first in the chain of credentials. Being first is needed in order to access open bucket that do not require a signature but the AWSCRT async client fails when that is first in the list.